frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlanMilestone/ReleasePlanMilestoneStatus.tsx

```diff
@@ -4,7 +4,11 @@ import PauseCircleIcon from '@mui/icons-material/PauseCircle';
 import TripOriginIcon from '@mui/icons-material/TripOrigin';
 import { useUiFlag } from 'hooks/useUiFlag';
 
-export type MilestoneStatus = 'not-started' | 'active' | 'paused' | 'completed';
+export type MilestoneStatus =
+    | { type: 'not-started'; scheduledAt?: Date }
```

Comment:
we explicitly model that only not-started milestones can be scheduled but don't have to be. This simplifies all child components that depend on scheduling